src/interceptors/Socket/MockSocket.ts

```diff
@@ -46,13 +50,6 @@ export class MockSocket extends net.Socket {
 
   public push(chunk: any, encoding?: BufferEncoding): boolean {
     this.options.read(chunk, encoding)
-
-    if (chunk !== null) {
-      this.emit('data', chunk)
-    } else {
-      this.emit('end')
-    }
-
-    return true
+    return super.push(chunk, encoding)
```

Comment:
Call `super.push()` for the `.push(null)` to correctly transition the socket into the end state:

- emit `end` event
- set `readable` to `false`
- set `readableEnded` to `true`

See: https://github.com/nodejs/node/blob/3a456c6db802b5b25594d3a9d235d4989e9f7829/lib/internal/streams/readable.js#L1696